packages/flutter/test/material/scaffold_test.dart

```diff
@@ -2031,6 +2031,36 @@ void main() {
       );
       await tester.pumpAndSettle();
     });
+
+    testWidgets('FloatingActionButton always keeps the same position regardless of extendBodyBehindAppBar', (WidgetTester tester) async {
+      await tester.pumpWidget(MaterialApp(
+        home: Scaffold(
+          appBar: AppBar(),
+          floatingActionButton: FloatingActionButton(
+            onPressed: () {},
+            child: const Icon(Icons.add),
+          ),
+          floatingActionButtonLocation: FloatingActionButtonLocation.endTop,
+          extendBodyBehindAppBar: false,
+        ),
+      ));
+      final Offset center0 = tester.getCenter(find.byType(FloatingActionButton));
```

Comment:
Just for clarity, can you rename these 'center0' 'center1' to indicate what state they represent? Like 'defaultOffset' & 'extendedBodyOffset'?

Reply:
@Piinks 
I renamed `center0` and `center1` to `defaultOffset` and `extendedBodyOffset`.
Thanks for your reply!